test/json_unittest.cc

```diff
@@ -0,0 +1,385 @@
+  });
+}
+
+TEST(TrivialToStringTest, False) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::boolean(false), "false");
+  });
+}
+
+
+// Trival Number
+
+TEST(TrivialParseTest, Number) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("2");
+    EXPECT_EQ(v->As<json::Number>()->value(), 2.0);
+  });
+}
+
+TEST(TrivialToStringTest, Number) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::number(2.0), "2.0");
+  });
+}
+
+
+// Trival String
+
+TEST(TrivialParseTest, StringEmpty) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("\"\"");
+    EXPECT_EQ(v->As<json::String>()->value().length(), 0);
+  });
+}
+
+TEST(TrivialToStringTest, StringEmpty) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::string(""), "\"\"");
+  });
+}
+
+TEST(TrivialParseTest, StringOneChar) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("\"x\"");
+    const auto& str = v->As<json::String>();
+    EXPECT_EQ(str->value(), "x");
+  });
+}
+
+TEST(TrivialToStringTest, StringOneChar) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::string("x"), "\"x\"");
+  });
+}
+
+
+// Trival Array
+
+TEST(TrivialParseTest, ArrayEmpty) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("[]");
+    EXPECT_TRUE(v->As<json::Array>()->empty());
+  });
+}
+
+TEST(TrivialToStringTest, ArrayEmpty) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::array({}), "[]");
+  });
+}
+
+TEST(TrivialParseTest, ArrayOneElement) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("[2]");
+    const auto& arr = v->As<json::Array>();
+    EXPECT_EQ(arr->size(), 1);
+    EXPECT_EQ((*arr)[0]->As<json::Number>()->value(), 2.0);
+  });
+}
+
+TEST(TrivialToStringTest, ArrayOneElement) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::array({json::number(2.0)}), "[2.0]");
+  });
+}
+
+
+// Trival Dict
+
+TEST(TrivialParseTest, DictEmpty) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("{}");
+    EXPECT_TRUE(v->As<json::Object>()->empty());
+  });
+}
+
+TEST(TrivialToStringTest, DictEmpty) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::object({}), "{}");
+  });
+}
+
+TEST(TrivialParseTest, DictOneField) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("{\"f\":2}");
+    const auto& obj = v->As<json::Object>();
+    EXPECT_EQ(obj->size(), 1);
+    EXPECT_EQ(obj->Get<json::Number>("f"), 2.0);
+  });
+}
+
+TEST(TrivialToStringTest, DictOneField) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::object({
+      {"f", json::number(2.0)}
+    });
+    EXPECT_TO_STRING(v, "{\"f\":2.0}");
+  });
+}
+
+
+// String edge cases
+
+/*
+ * TODO(igorp): Seems that yajl does not support valid JSON.
+TEST(EdgeTest, StringWithNewLine) {
```

Comment:
Nit: "newline" is usually one word, so `StringWithNewline`?

Reply:
Done.